### ISSUE TITLE
Quad SPI mode 

### ISSUE THREAD

**Issue by 0x0fe** [espressif/arduino-esp32#7063]
### Related area

adding QSPI mode to the SPI driver

### Hardware specification

ESP32, ESP32C3

### Is your feature request related to a problem?

I could not find any implementation of QSPI mode in the ESP32 core. QSPI is useful in various scenarios, flash, ADCs/DACs, fpgas, inter chip communiation, custom interfaces.

### Describe the solution you'd like

Being able to select the QSPI mode on a SPI interface, as it has been done here, https://github.com/CutClassH/sea_esp32_qspi but this does not work anymore.

### Describe alternatives you've considered

in my case i want to send 3 parallele 8 bit streams, that is, sending 3 separate bistreams on 3 GPIOs synced with the same clock.QSPI seems like the best way to do it at decent speed. An alternative is to bitbang, the last recourse, but bitbanging on ESP32 is not really practical (by experience).

### I have checked existing list of Feature requests and the Contribution Guide

- [X] I confirm I have checked existing list of Feature requests and Contribution Guide.

**Comment by Jason2866** [comment]
The connected flash (where the sketch is flashed) to the ESP32, C3, S2 can use mode qio. Or do you mean something different?

**Comment by 0x0fe** [comment]
no, i mean for the SPI driver of arduino core, the "user side" driver. Currently it does not support QSPI (at least on C3).

**Comment by 0x0fe** [comment]
if you check the initial message i think it was qquite clear though, with example linked

**Comment by VojtechBartoska** [comment]
Related request from the discussion: https://github.com/espressif/arduino-esp32/discussions/8174

**Comment by 0x0fe** [comment]
@VojtechBartoska  OK, well noted

**Comment by kylongmu** [comment]
> Postponing this to 3.1.0 Milestone, we need to evaluate this again.

Waiting for the qspi support in arduino too.

**Comment by xChumimin** [comment]
I need more documentation, examples and support on both qspi and dual io spi

**Comment by Parsaabasi** [comment]
Hello, 

Due to the overwhelming volume of issues currently being addressed, we have decided to close the previously received tickets. If you still require assistance or if the issue persists, please don't hesitate to reopen the ticket.

Thanks.
